### DISCUSSION TITLE
Execution of resolver chains

### DISCUSSION THREAD

**Discussion by DimuthuMadushan** [apollographql/apollo-server#6641] [Q&A]
This is regarding the resolver chain in the apollo server. I’m new to apollo GraphQL and I need to clarify the following. Let’s say we have a GraphQL field that returns an Object. Assume that object has more than one field and the query asks for one of them. When executing the resolver for that field, the resolver argument `parent` included the complete result of the parent object even if the query is asked for one field. My problem is isn’t this inefficient that resolves all the object fields rather than resolving the field query asks for? Is there any special reason for this behavior?

**Comment by glasser** [comment]
Given

```
type Foo {
  bar: ID
}

type Query {
  foo: Foo
}
```

When you run `{ foo { bar } }` the parent argument passed to the Foo.bar resolver is whatever you personally choose to return from the Query.foo resolver.

If your implementation of `Query.foo` is "do a DB lookup with a zillion fields and keep them all in memory" then sure, you'll have lots of fields on that object. But that's your choice!  Your Query.foo doesn't even have to return an object at all: eg, it could just return an ID string, and resolvers like `Foo.bar` would receive that string as the parent argument.
